File: frontend/services/trust-score.ts
import { api, type ApiEnvelope } from '@/lib/api';
import type { ListQueryState, PaginatedList } from '@/lib/pagination/types';

export type TrustScoreLevel = 'gold' | 'silver' | 'bronze' | 'high_risk';

export const TRUST_LEVEL_LABELS: Record<TrustScoreLevel, string> = {
  gold: 'Gold',
  silver: 'Silver',
  bronze: 'Bronze',
  high_risk: 'High Risk',
};

export interface TrustScoreEvent {
  id: string;
  user_id: string;
  event_type: string;
  label: string;
  delta: number;
  score_before: number;
  score_after: number;
  reference_type: string | null;
  reference_id: string | null;
  metadata: Record<string, unknown> | null;
  created_at: string;
}

export interface CustomerTrustScoreSummary {
  user_id: string;
  full_name: string;
  email: string | null;
  phone: string | null;
  role: string;
  trust_score: number;
  level: TrustScoreLevel;
  level_label: string;
  risk_level: string | null;
  delivered_orders: number;
  dispute_count: number;
  refund_count: number;
  status: string;
  created_at: string;
}

export interface CustomerTrustScoreDetail extends CustomerTrustScoreSummary {
  events: TrustScoreEvent[];
}

export interface TrustScoreListFilters {
  role?: string;
  risk_level?: string;
  trust_score_min?: number;
  trust_score_max?: number;
  status?: string;
  created_from?: string;
  created_to?: string;
}

export type TrustScoreListParams = ListQueryState & TrustScoreListFilters;

export async function listAdminTrustScores(
  params: TrustScoreListParams = {},
): Promise<PaginatedList<CustomerTrustScoreSummary>> {
  const { data } = await api.get<ApiEnvelope<PaginatedList<CustomerTrustScoreSummary>>>(
    '/admin/trust-scores',
    { params },
  );
  return data.data;
}

export async function getAdminTrustScoreDetail(userId: string): Promise<CustomerTrustScoreDetail> {
  const { data } = await api.get<ApiEnvelope<CustomerTrustScoreDetail>>(`/admin/trust-scores/${userId}`);
  return data.data;
}

File: backend/app/schemas/trust_score.py
"""Customer trust score API schemas."""

from __future__ import annotations

from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.models.enums import TrustScoreEventType, TrustScoreLevel

TRUST_SCORE_EVENT_LABELS: dict[TrustScoreEventType, str] = {
    TrustScoreEventType.refund_request: "Refund Request",
    TrustScoreEventType.dispute_filed: "Dispute Filed",
    TrustScoreEventType.chargeback: "Chargeback",
    TrustScoreEventType.failed_payment: "Failed Payment",
    TrustScoreEventType.fake_claim: "Fake Claim (Rejected Dispute)",
    TrustScoreEventType.successful_order: "Successful Order",
    TrustScoreEventType.positive_review: "Positive Review",
    TrustScoreEventType.long_history: "Long Customer History",
}

TRUST_LEVEL_LABELS: dict[TrustScoreLevel, str] = {
    TrustScoreLevel.gold: "Gold",
    TrustScoreLevel.silver: "Silver",
    TrustScoreLevel.bronze: "Bronze",
    TrustScoreLevel.high_risk: "High Risk",
}


class TrustScoreEventResponse(BaseModel):
    id: UUID
    user_id: UUID
    event_type: TrustScoreEventType
    label: str
    delta: int
    score_before: int
    score_after: int
    reference_type: str | None
    reference_id: UUID | None
    metadata: dict[str, Any] | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class CustomerTrustScoreSummary(BaseModel):
    user_id: UUID
    full_name: str
    email: str | None
    phone: str | None = None
    role: str = "customer"
    trust_score: int
    level: TrustScoreLevel
    level_label: str
    risk_level: str | None = None
    delivered_orders: int
    dispute_count: int = 0
    refund_count: int = 0
    status: str = "active"
    created_at: datetime


class CustomerTrustScoreDetail(CustomerTrustScoreSummary):
    events: list[TrustScoreEventResponse]
